oe_contact_forms.module

```diff
@@ -6,3 +6,335 @@
+ *
+ * @param array $form
+ *   The form.
+ * @param Drupal\Core\Form\FormStateInterface $form_state
+ *   The form state.
+ *
+ * @return array
+ *   Topics fieldset.
+ */
+function _oe_contact_forms_add_more_callback(array &$form, FormStateInterface $form_state): array {
+  return $form['corporate_fields']['topics_fieldset'];
+}
+
+/**
+ * Add element to the form based on ajax.
+ *
+ * @param array $form
+ *   The form.
+ * @param Drupal\Core\Form\FormStateInterface $form_state
+ *   The form state.
+ */
+function _oe_contact_forms_add_one_callback(array &$form, FormStateInterface $form_state): void {
+  $num_topics = $form_state->get('num_topics');
+  $num_topics++;
+  $form_state->set('num_topics', $num_topics);
+  $form_state->setRebuild();
+}
+
+/**
+ * Remove element from the form based on ajax.
+ *
+ * @param array $form
+ *   The form.
+ * @param Drupal\Core\Form\FormStateInterface $form_state
+ *   The form state.
+ */
+function _oe_contact_forms_remove_callback(array &$form, FormStateInterface $form_state): void {
+  $num_topics = $form_state->get('num_topics');
+
+  if ($num_topics > 1) {
+    $num_topics--;
+  }
+
+  $form_state->set('num_topics', $num_topics);
+  $form_state->setRebuild();
+}
+
+/**
+ * Ajax callback: Return markup for 'corporate_fields'.
+ *
+ * According 'is_corporate_form' value.
+ *
+ * @param array $form
+ *   The form.
+ * @param Drupal\Core\Form\FormStateInterface $form_state
+ *   The form state.
+ *
+ * @return array
+ *   Corporate fields.
+ */
+function _oe_contact_forms_is_corporate_form_ajax_callback(array &$form, FormStateInterface $form_state): array {
+  return $form['corporate_fields'];
+}
+
+/**
+ * Entity builder for the contact form edit form with third party options.
+ *
+ * @param string $entity_type
+ *   The contact form entity.
+ * @param \Drupal\contact\ContactFormInterface $contact_form
+ *   The contact form entity.
+ * @param array $form
+ *   The form.
+ * @param Drupal\Core\Form\FormStateInterface $form_state
+ *   The form state.
+ */
+function _oe_contact_forms_contact_form_builder($entity_type, ContactFormInterface $contact_form, array &$form, FormStateInterface $form_state): void {
+  // When ajax is triggered the form is rebuilt with empty recipients,
+  // causing invalid arguments in Drupal\contact\ContactFormEditForm::form()L96.
+  $recipients = $contact_form->getRecipients();
+
+  if (!is_array($recipients)) {
+    $contact_form->setRecipients((array) $recipients);
+  }
+
+  $is_corporate_form = (boolean) $form_state->getValue('is_corporate_form');
+  $values = $form_state->getValue('corporate_fields');
+  $triggered_element = $form_state->getTriggeringElement();
+
+  if (!empty($triggered_element['#submit']) && in_array('_oe_contact_forms_add_one_callback', $triggered_element['#submit'])) {
+    $contact_form->unsetThirdPartySetting('oe_contact_forms', 'topics');
+    $is_corporate_form = FALSE;
+  }
+
+  if ($is_corporate_form && !empty($values)) {
```

Comment:
There is a problem here as it doesn't allow to alter a form that has been set to be corporate to behave as non-corporate. In case $is_corporate_form is FALSE we should still update the 3rdPartySetting for "is_corporate_form". Tests should be also added for this situation.

Reply:
fixed